apic_ml2/neutron/plugins/ml2/drivers/cisco/apic/mechanism_apic.py

```diff
@@ -1470,6 +1520,27 @@ def create_network_postcommit(self, context):
             self.apic_manager.update_name_alias(
                 self.apic_manager.apic.fvAEPg, tenant_id, app_profile_name,
                 epg_name, nameAlias=network_name)
+            # set the tenant or application profile nameAlias if this is the
+            # first network being created under this tenant
```

Comment:
actually discussed similar things with Mandeep yesterday, especially for GBP workflow. In GBP, user can create l3p, l2p, PRS then ptg for example. But we don't want to do this in all the places so in GBP we will just do it when dealing with PTG. So yes, there could be a delay (the nameAlias field won't get set until PTG is created) but Mandeep is ok with it since without PTG, nothing will work. But in ML2 mode, my understanding is that when you create the router, those contracts are being created under tenant common though. But anyway the same concept can apply here in ML2 if this is not the case.